### PR TITLE
Adds the gifted "negative" quirk

### PR DESCRIPTION
**Pull request by TheSS13Melon**
## About The Pull Request

JungleRat wanted a quirk that gave some freebie points. This perk, gifted, gives the player 6 free negative points to use on positive quirk points.

## How This Contributes To The Skyrat Roleplay Experience

Ultimately this allows more player expression without taking debilitating quirks.

## Proof of Testing

<details>
<summary>

</summary>
  
</details>

## Changelog

:cl:
add: Adds a "negative" perk called gifted which gives 6 free points.
/:cl:

### CODE REVIEW

**Comment by burgerenergy** [comment]
could

we not just non-modular edit in 4-6 more points

that said it should be obvious in the blurb that it doesn't do anything

**Comment by Stalkeros2** [comment]
Wtf based???

**Comment by W4rd3nn** [comment]
time to be a med doc and scan and bully everyone who took this

**Comment by Iamgoofball** [comment]
just add some freebie points OP this is a bad way to do it

File: code/_globalvars/traits.dm [hunk 1]
```diff
@@ -153,6 +153,7 @@ GLOBAL_LIST_INIT(traits_by_type, list(
 		"TRAIT_ALCOHOL_TOLERANCE" = TRAIT_ALCOHOL_TOLERANCE,
 		"TRAIT_AGEUSIA" = TRAIT_AGEUSIA,
 		"TRAIT_HEAVY_SLEEPER" = TRAIT_HEAVY_SLEEPER,
+		"TRAIT_GIFTED" = TRAIT_GIFTED,
```
Comment (vinylspiders) [inline code/_globalvars/traits.dm]:
```suggestion
		"TRAIT_GIFTED" = TRAIT_GIFTED, // SKYRAT EDIT ADDITION
```

File: code/datums/quirks/negative_quirks/gifted.dm [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/datum/quirk/gifted
+	name = "Gifted"
+	desc = "You were born a bit lucky, intelligent, or something in between. You're able to do a little more."
+	icon = FA_ICON_DOVE
+	value = -6
+	mob_trait = TRAIT_GIFTED
+	gain_text = span_danger("You feel like you're just a little bit more flexible.")
+	lose_text = span_notice("You feel a little less flexible.")
+	medical_record_text = "Patient has a history of uncanny fortune."
+	hardcore_value = 0
```
Comment (vinylspiders) [inline code/datums/quirks/negative_quirks/gifted.dm]: Move this to modular_skyrat\master_files\code\datums\quirks\negative.dm

**Review by vinylspiders — CHANGES_REQUESTED** [review]
Good otherwise, refer to the modularization guide [here](https://github.com/Skyrat-SS13/Skyrat-tg/blob/master/modular_skyrat/readme.md) if you're not sure what I'm talking about with these comments.

We try to keep the Skyrat code separate from TG code is the short of it.

**Comment by vinylspiders** [comment]
All said though, if you are able to, as others have said it would be better to just add 6 points to the starting balance. But I think it might be slightly tricky due to the very odd way that quirk points are coded. 

You will likely have to make some tgui edits in addition to modifying `getQuirkBalance()` to start at -6 points (yes -6, the sign gets flipped after being passed to tgui...)

You can probably get some help from the contributor chat if you need it.

**Comment by HoodMilk** [comment]
We should just make all quirks have 0 cost, you can still only take 6 positive ones anyway. Would allow people to take negative quirks for roleplaying reasons rather than "I need to tg balance my character".

**Comment by TheSS13Melon** [comment]
> All said though, if you are able to, as others have said it would be better to just add 6 points to the starting balance. But I think it might be slightly tricky due to the very odd way that quirk points are coded.
> 
> You will likely have to make some tgui edits in addition to modifying `getQuirkBalance()` to start at -6 points (yes -6, the sign gets flipped after being passed to tgui...)
> 
> You can probably get some help from the contributor chat if you need it.

This is how junglerat wanted it done and iirc this is what was approved by the maintainers he got to sign off on it. I'd bring it up with him and the maintainers if we want to have it done like this.

**Comment by lessthnthree** [comment]
> > All said though, if you are able to, as others have said it would be better to just add 6 points to the starting balance. But I think it might be slightly tricky due to the very odd way that quirk points are coded.
> > You will likely have to make some tgui edits in addition to modifying `getQuirkBalance()` to start at -6 points (yes -6, the sign gets flipped after being passed to tgui...)
> > You can probably get some help from the contributor chat if you need it.
> 
> This is how junglerat wanted it done and iirc this is what was approved by the maintainers he got to sign off on it. I'd bring it up with him and the maintainers if we want to have it done like this.

That... is the maintainer.

**Comment by TheSS13Melon** [comment]
> > > All said though, if you are able to, as others have said it would be better to just add 6 points to the starting balance. But I think it might be slightly tricky due to the very odd way that quirk points are coded.
> > > You will likely have to make some tgui edits in addition to modifying `getQuirkBalance()` to start at -6 points (yes -6, the sign gets flipped after being passed to tgui...)
> > > You can probably get some help from the contributor chat if you need it.
> > 
> > 
> > This is how junglerat wanted it done and iirc this is what was approved by the maintainers he got to sign off on it. I'd bring it up with him and the maintainers if we want to have it done like this.
> 
> That... is the maintainer.

He didn't specify who signed off on it. Just that two maintainers signed off.

**Comment by lessthnthree** [comment]
> > > > All said though, if you are able to, as others have said it would be better to just add 6 points to the starting balance. But I think it might be slightly tricky due to the very odd way that quirk points are coded.
> > > > You will likely have to make some tgui edits in addition to modifying `getQuirkBalance()` to start at -6 points (yes -6, the sign gets flipped after being passed to tgui...)
> > > > You can probably get some help from the contributor chat if you need it.
> > > 
> > > 
> > > This is how junglerat wanted it done and iirc this is what was approved by the maintainers he got to sign off on it. I'd bring it up with him and the maintainers if we want to have it done like this.
> > 
> > 
> > That... is the maintainer.
> 
> He didn't specify who signed off on it. Just that two maintainers signed off.

I'm the other one. So I suggest you take [vinylspiders](https://github.com/vinylspiders)'s suggestion.

**Comment by vinylspiders** [comment]
> > > > All said though, if you are able to, as others have said it would be better to just add 6 points to the starting balance. But I think it might be slightly tricky due to the very odd way that quirk points are coded.
> > > > You will likely have to make some tgui edits in addition to modifying `getQuirkBalance()` to start at -6 points (yes -6, the sign gets flipped after being passed to tgui...)
> > > > You can probably get some help from the contributor chat if you need it.
> > > 
> > > 
> > > This is how junglerat wanted it done and iirc this is what was approved by the maintainers he got to sign off on it. I'd bring it up with him and the maintainers if we want to have it done like this.
> > 
> > 
> > That... is the maintainer.
> 
> He didn't specify who signed off on it. Just that two maintainers signed off.

We never really 'signed off' on this specific way of doing it, just kind of a general agreement that having more quirk points to play with would be a good idea. 

I think this way is a bit cumbersome and one of the concerns I had was people being grandfathered into having an extra 6 points worth of quirks should we ever remove this.

I will say, if you don't feel confident in being able to come up with a cleaner way then I'll accept this I guess. Still not a fan though.

File: code/_globalvars/traits.dm [hunk 3]
```diff
@@ -153,6 +153,7 @@ GLOBAL_LIST_INIT(traits_by_type, list(
 		"TRAIT_ALCOHOL_TOLERANCE" = TRAIT_ALCOHOL_TOLERANCE,
 		"TRAIT_AGEUSIA" = TRAIT_AGEUSIA,
 		"TRAIT_HEAVY_SLEEPER" = TRAIT_HEAVY_SLEEPER,
+		"TRAIT_GIFTED" = TRAIT_GIFTED, // skyrat edit addition
```
Comment (vinylspiders) [inline code/_globalvars/traits.dm]:
```suggestion
		"TRAIT_GIFTED" = TRAIT_GIFTED, // SKYRAT EDIT - ADDITION
```

**Review by vinylspiders — CHANGES_REQUESTED** [review]

File: code/__DEFINES/~skyrat_defines/traits.dm [hunk 4]
```diff
@@ -17,6 +17,8 @@
 /// This makes trait makes it so that the person cannot be infected by the zombie virus.
 #define TRAIT_MUTANT_IMMUNE "mutant_immune"
 
+#define TRAIT_GIFTED "gifted" // adds -6 quirk to negative quirks for free points.
```
Comment (vinylspiders) [inline code/__DEFINES/~skyrat_defines/traits.dm]:
```suggestion
/// adds -6 quirk to negative quirks for free points.
#define TRAIT_GIFTED "gifted"
```
autodoc

**Review by vinylspiders — APPROVED** [review]
After speaking on discord, I'm ok with this solution given the circumstances. 

If anyone wants to do a cleaner solution that modifies tgui they're welcome to do so at any point.

File: modular_skyrat/master_files/code/datums/quirks/negative.dm [hunk 5]
```diff
@@ -86,3 +86,14 @@
 	var/datum/quirk/blooddeficiency/synth/bd_synth = new
 	qdel(src)
 	return bd_synth.add_to_holder(new_holder, quirk_transfer)
+
+/datum/quirk/gifted
+	name = "Gifted"
+	desc = "You were born a bit lucky, intelligent, or something in between. You're able to do a little more."
+	icon = FA_ICON_DOVE
+	value = -6
```
Comment (vinylspiders) [inline modular_skyrat/master_files/code/datums/quirks/negative.dm:95]:
```suggestion
	quirk_flags = QUIRK_HUMAN_ONLY|QUIRK_HIDE_FROM_SCAN
	value = -6
```
One more thing, this probably shouldn't show up on scans.

**Pull request merged**